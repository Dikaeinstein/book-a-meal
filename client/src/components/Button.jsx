import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ value, className }) => (
  <button className={className}>
    {value}
  </button>
);

Button.propTypes = {
  value: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
};

export default Button;
