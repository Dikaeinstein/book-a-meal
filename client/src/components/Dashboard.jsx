import React from 'react';
import OrderHistory from './OrderHistory';

const Dashboard = () => <h2>Dashboard</h2>;

export default Dashboard;
