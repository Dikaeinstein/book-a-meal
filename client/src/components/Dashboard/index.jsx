import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import DashboardOrderHistory from './DashboardOrderHistory';
import Footer from '../util/Footer';
import FilterOrders from './FilterOrders';
import TotalOrders from './TotalOrders';
import TotalSales from './TotalSales';
import errorHandler from '../util/errorHandler';


const ConnectedDashboard = ({ fetchOrdersError }) => {
  const DashboardOrderHistoryWithErrorHandling =
    errorHandler(DashboardOrderHistory, 'Error fetching orders');

  return (
    <div>
      <main
        className="dash-main text-center bg-dark-light"
        style={{ minHeight: 'calc(100vh - 151px)' }}
      >
        <h1>Dashboard</h1>
        <section className="card dashboard">
          <TotalOrders />
          <TotalSales />
          <FilterOrders />
          <DashboardOrderHistoryWithErrorHandling
            error={fetchOrdersError}
          />
        </section>
      </main>
      <Footer />
    </div>
  );
};

ConnectedDashboard.propTypes = {
  /* eslint react/require-default-props: 0 */
  fetchOrdersError: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.objectOf(PropTypes.string),
  ]),
};

const mapStateToProps = state => ({
  fetchOrdersError: state.orders.error,
});

const Dashboard = connect(mapStateToProps)(ConnectedDashboard);

export default Dashboard;
