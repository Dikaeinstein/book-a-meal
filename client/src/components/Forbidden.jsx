import React from 'react';
import { NavLink } from 'react-router-dom';
import Footer from './util/Footer';

const Forbidden = () => (
  <main>
    <div
      className="forbidden-overlay"
      style={{
        minHeight: 'calc(100vh - 145px)',
        padding: '1rem',
      }}
    >
      <h6
        className="text-center"
        style={{ paddingTop: '60vh' }}
      >
        Sorry, access to this page is denied.
        Either check the url or <NavLink to="/">go home</NavLink>
      </h6>
    </div>
    <Footer />
  </main>
);

export default Forbidden;
