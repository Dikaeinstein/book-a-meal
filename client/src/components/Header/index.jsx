import React from 'react';
import PropTypes from 'prop-types';
import { ToastContainer } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.min.css';
import Logo from './Logo';
import Navigation from './Navigation';

const Header = ({ className }) => (
  <header className={className}>
    <ToastContainer />
    <Logo
      text="Book-A-Meal"
      className="col-1-4 logo text-left text-dark"
    />
    <div className="toggle-show">&#9776;</div>
    <Navigation />
  </header>
);

Header.propTypes = {
  className: PropTypes.string.isRequired,
};

export default Header;
