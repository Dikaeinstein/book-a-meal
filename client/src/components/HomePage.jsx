import React from 'react';
import IntroBanner from './IntroBanner';
import MenuCard from './MenuCard';

const HomePage = () => (
  <div>
    <IntroBanner />
    <MenuCard link="/signin" />
  </div>
);

export default HomePage;
