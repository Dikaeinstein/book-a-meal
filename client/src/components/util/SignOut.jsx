import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { removeUser } from '../../helpers/persistUser';
import { userSignOut } from '../../actions/userActions';

export class SignOut extends Component {
  componentDidMount() {
    userSignOut();
    removeUser();
  }

  render() {
    return <React.Fragment>{this.props.children}</React.Fragment>;
  }
}

SignOut.propTypes = {
  children: PropTypes.element.isRequired,
};

export default connect(null, { userSignOut })(SignOut);
