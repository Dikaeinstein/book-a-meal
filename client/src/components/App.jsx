import React, { Component } from 'react';
import Header from './Header';
import IntroBanner from './IntroBanner';
import MenuCard from './MenuCard';
import Footer from './Footer';

const App = () => (
  <div>
    <Header className="navbar" />
    <IntroBanner />
		<MenuCard />
    <Footer />
  </div>
);

export default App;
