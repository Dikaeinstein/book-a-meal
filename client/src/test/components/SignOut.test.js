import React from 'react';
import { shallow } from 'enzyme';
import { SignOut } from '../../components/SignOut';

describe('<SignOut />', () => {
  it('should match snapshot', () => {
    const tree = shallow(<SignOut><div>test</div></SignOut>);
    expect(tree).toMatchSnapshot();
  });
});
