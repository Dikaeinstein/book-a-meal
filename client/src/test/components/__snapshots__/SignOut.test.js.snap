// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignOut /> should match snapshot 1`] = `
<React.Fragment>
  <div>
    test
  </div>
</React.Fragment>
`;
